dio_test/lib/src/test/download_tests.dart

```diff
@@ -380,6 +380,54 @@ void downloadTests(
           completes,
         );
       });
+      test('append bytes previous download', () async {
+        final cancelToken = CancelToken();
+        final path = p.join(tmp.path, 'download_3.txt');
+        final requestedBytes = 1024 * 1024 * 10;
+        var recievedBytes1 = 0;
+        await expectLater(
+          dio.download(
+            '/bytes/$requestedBytes',
+            path,
+            cancelToken: cancelToken,
+            onReceiveProgress: (c, t) {
+              if (c > 5000) {
```

Comment:
The test seems fake when it involves the real world's use case. Predicting the length of bytes is meanless since:
1. You've set the condition to cancel when bytes reach 5000.
2. No actual content has been validated.

Reply:
canceling the request is mandatory for this test to work because I am cancelling the first download request after loading 5000 bytes and make the second request to verify adding to existing file we are requesting 1024*1024*10 
we are loading 5000 first then 5000 as appending in the check we are checking if the summation of the two first requests equals file size and the test confirms this

Reply:
What I mean is validating the length cannot prove the content is being written correctly. Say you have a content sequence `abcdefg`, if you wrote `efgabcd` which is the same length, the length comparison is then fake.

Reply:
validating the sequence of content is not our job we just need to validate that the bytes are added to the existed file, We just enabled developer to change file access mode.

We need to validate bytes order if we are making a complete feature of resuming the download including checking the existing file size and requesting a range size.